File: algorithms/dichotomy/dichotomy.py
array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

def binary_search(array, item):
    low_index = 0
    heigh_index = len(array) - 1

    while low_index <= heigh_index:
        mid_index = (low_index + heigh_index) / 2
        mid_index = int(mid_index)
        guess_value = array[mid_index]
        if item == guess_value:
            return mid_index
        if item < guess_value:
            heigh_index = mid_index - 1
        else:
            low_index = mid_index + 1
    return None

result = binary_search(array, 3)
print(array[result])

File: algorithms/dichotomy/dichotomy.c
#include <stdio.h>

int binary_search(int[], int);

int main() {
  int array[] = {1, 2, 3, 4, 5, 6, 7, 8};
  int result = binary_search(array, 8);
  printf("%d\n", result);
}

int binary_search(int array[], int target) {
  int low_index = 0;
  int heigh_index = sizeof(array);
  while (low_index <= heigh_index) {
    int mid_index = (low_index + heigh_index) / 2;
    int guess_value = array[mid_index];
    if (target == guess_value) {
      return mid_index;
    }
    if (target < guess_value) {
      heigh_index = mid_index - 1;
    } else {
      low_index = mid_index + 1;
    }
  }
}
